### DISCUSSION TITLE
C# how to configure SPFx webpart searchable properties

### DISCUSSION THREAD

**Discussion by patrickblanc** [pnp/pnpframework#449] [Q&A] [answered]
I am adding a webpart like this :
```c
var component = page.AvailablePageComponents().FirstOrDefault(c => new Guid(c.Id) == new Guid("GUID"));
if (component != null)
{
    var webpart = page.NewWebPart(component);
    AccordionProperties properties = new ()
    {
        Description = "Accordion",
        AccordionItems = items,
    };

    webpart.PropertiesJson = JsonSerializer.Serialize(properties, new JsonSerializerOptions()
    {
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
        WriteIndented = true,
    });
    page.AddControl(webpart, page.Sections[0].Columns[0]);
}
```

The SPFx webpart set the properties as searchable with the following configuration : 

```typescript
protected get propertiesMetadata(): IWebPartPropertiesMetadata {
  return {
    'accordionItems[*].header': { isSearchablePlainText: true },
    'accordionItems[*].content': { isSearchablePlainText: true },
  }
}
```

My problem is the following, on the webpart, we can't set webpart.**ServerProcessedContent** by code and it seems not automatically done during the process. So the webpart in the page looks like this : 

![image](https://user-images.githubusercontent.com/70770642/129909517-85416f78-5617-466d-b463-bfc6e942ba3f.png)

If I search the text within my webpart in the search at this moment, no result is found.

But if I click on the edit button on the modern page and save it directly... then the properties is empty, but the serverProcessedContent contains my items and it is now searchable.

![image](https://user-images.githubusercontent.com/70770642/129909689-e3f22798-1a10-4dff-b2be-43cf1c5c595b.png)

Then if I am searching for the webpart text content, result start appearing in the search.

**Comment by jansenbe** [comment, marked as answer]
Asnwered in https://github.com/pnp/pnpframework/issues/451
